examples/images_contours_and_fields/colorbar_annotation_demo.py

```diff
@@ -0,0 +1,54 @@
+"""
+========================
+Colorbar Annotation Demo
+========================
+
+Demonstrates how one can annotate the colorbar without needing to also add
+lines to a contour plot.
+
+Note that this is not a typical use case. The more common use case for
+annotations is to add lines to both a contour plot *and* the corresponding
+location on the colorbar. In this more common use case,
+`~matplotlib.colorbar.Colorbar.add_lines` can be used.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Create sample values to plot
+x = np.linspace(0, 1, 100)
+y = np.linspace(0, 1, 100)
+X, Y = np.meshgrid(x, y)
+Z = np.exp(-X**2 - Y**2)
+
+# Choose a value to annotate
+desired_cbar_yvalue = 0.90
+
+# Build filled contour plot
+surfplot = plt.contourf(X, Y, Z, cmap=plt.cm.binary_r)
+cbar = plt.colorbar(surfplot)
+
+# Find min and max values on colorbar
+cmin = np.float(cbar.ax.get_yticklabels()[0].get_text())
+cmax = np.float(cbar.ax.get_yticklabels()[-1].get_text())
+
+# Use min/max values to calculate annotation location
+loc = (desired_cbar_yvalue - cmin) / (cmax - cmin)
```

Comment:
I agree that you need to do this for `contour` colorbars, but its not true for `pcolormesh` colorbars.  So there needs to be more explanation here.  And frankly, I think the correct thing is for contour colorbars to be rendered between vmin and vmax.  So I'm a little leery about adding an example that shows how to get around inconsistent behaviour.

Reply:
Thanks! I can certainly add more explanation here (depending perhaps on what we decide to do with this example, given the discussion below).